presto-main/src/test/java/com/facebook/presto/operator/TestWorkProcessorUtils.java

```diff
@@ -31,10 +31,37 @@
 import static java.util.Objects.requireNonNull;
 import static org.testng.Assert.assertEquals;
 import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertThrows;
 import static org.testng.Assert.assertTrue;
 
 public class TestWorkProcessorUtils
 {
+    @Test
+    public void testIterator()
+    {
+        WorkProcessor<Integer> processor = processorFrom(ImmutableList.of(
+                ProcessorState.ofResult(1),
+                ProcessorState.ofResult(2),
+                ProcessorState.finished()));
+
+        Iterator<Integer> iterator = processor.iterator();
```

Comment:
Since you're testing `WorkProcessor#iterator`, shouldn't this be in the `TestWorkProcessor` class? (vs `TestWorkProcessorUtils`)

Reply:
I didn't want to clutter `WorkProcessor` interface, therefore actual code is located in `WorkProcessorUtils`. Because of that, corresponding tests are in `TestWorkProcessorUtils`